src/main/java/com/scmp/framework/testng/listeners/SuiteListener.java

```diff
@@ -180,4 +190,91 @@ private void createNewTestRun(ISuite suite, String projectId, String testRunName
+	 * 2. For each test run, it will filter the failed case
+	 * 3. Filter out which test cases are not in all test runs and remove it from the final list
+	 * 4. Filter out which test cases are in the exclude list and remove it from the final list
+	 * 5. Send the final list to Slack channel
+	 */
+	private void logConsecutiveFailedTestCase() {
+
+		String projectId = frameworkConfigs.getTestRailProjectId();
+
+		// Timestamp x days before set in the configuration
+		LocalDate today = LocalDate.now(runTimeContext.getZoneId());
+		String timestamp = String.valueOf(today.minusDays(frameworkConfigs.getFailedCaseTestRunWithinDays()).atStartOfDay(runTimeContext.getZoneId()).toEpochSecond());
+		HashSet<Integer> failedTestCasesIdSet = new HashSet<>();
+		boolean isFirstRunLoop = true;
+		Map<Integer, String> finalResultMap = new HashMap<>();
+
+		try {
+
+			Pattern matchTestRunPattern = Pattern.compile(frameworkConfigs.getFailedCaseTestRunNotificationPattern());
+
+			// Get the latest x test runs and match with pattern set in configuration
+			List<TestRun> runs = testRailManager.getTestRuns(projectId, timestamp).getTestRunList().stream()
+					.filter(testRun -> matchTestRunPattern.matcher(testRun.getName()).find())
+					.limit(frameworkConfigs.getFailedCaseNotificationCount())
+					.toList();
+
+			// Search for consecutive failed test cases
+			for (TestRun run : runs) {
+
+				// Skip listing if the test run is in progress
+				if (!testRailManager.getAllTestRunTests(run.getId(), String.valueOf(TestRailStatus.IN_PROGRESS)).isEmpty()) {
+					return;
+				}
+
+				// Get all failed test cases
+				Map<Integer, String> runResult = testRailManager.getAllTestRunTests(run.getId(), String.valueOf(TestRailStatus.Failed))
+						.stream()
+						.collect(Collectors.toMap(TestRunTest::getCaseId, TestRunTest::getTitle));
+
+				if (isFirstRunLoop) {
+					failedTestCasesIdSet.addAll(runResult.keySet());
+					finalResultMap.putAll(runResult);
+					isFirstRunLoop = false;
+				} else {
+					failedTestCasesIdSet.retainAll(runResult.keySet());
+				}
+			}
+
+			// Remove those test that are failed first time only but not consecutive times
+			finalResultMap.keySet().removeIf(testId -> !failedTestCasesIdSet.contains(testId));
+
+			// Remove test case in exclude list
+			HashSet<String> excludeTestCase = new HashSet<>(Arrays.asList(frameworkConfigs.getFailedCaseExcludeList().split(",")));
+			finalResultMap.keySet().removeIf(testId -> excludeTestCase.contains(String.valueOf(testId)));
+
+			// Display list
+			if (!finalResultMap.isEmpty()) {
+				frameworkLogger.info("Consecutive failed test cases: ");
+				finalResultMap.forEach((k, v) -> frameworkLogger.info("Test Case ID: {}, Title: {}", k, v));
+			} else {
+				frameworkLogger.info("No consecutive failed test cases found.");
```

Comment:
should we return here if no failed cases?

Reply:
Added return